### PR TITLE
feat: use sessiond mconfig to initialize sentry in sessiond

### PR DESCRIPTION
**Pull request by waqaraqeel**
Signed-off-by: Waqar Aqeel <waqeel@fb.com>

## Summary
Working towards #7613.

This PR makes changes to sessiond sentry initialization so that relevant configuration sent from orc8r are used. Configurations in `control_proxy.yml` still take precedence over those sent from orc8r.

## Test Plan

Ran `make test_session_manager` in `lte/gateway` with `SENTRY_ENABLED` flag set.

### CODE REVIEW

**Review by themarwhal — COMMENTED** [review]
Can we modify `lte/gateway/configs/control_proxy.yml` so that the default sample rate is 1?

File: lte/gateway/c/session_manager/SentryWrappers.cpp [hunk 1]
```diff
@@ -37,33 +38,56 @@ using std::experimental::optional;
...
 // TODO(@themarwhal) pull common sentry functions into lib common
 
-optional<std::string> get_sentry_url(YAML::Node control_proxy_config) {
-  std::string sentry_url;
+optional<std::string> get_sentry_url(
+    const magma::mconfig::SentryConfig& sentry_config,
+    YAML::Node control_proxy_config) {
   if (control_proxy_config[SENTRY_NATIVE_URL].IsDefined()) {
-    const std::string sentry_dns =
+    const std::string override =
```
Comment (themarwhal) [inline lte/gateway/c/session_manager/SentryWrappers.cpp]: `dns_override` just to be extra clear

File: lte/gateway/c/session_manager/SentryWrappers.cpp [hunk 2]
```diff
@@ -37,33 +38,56 @@ using std::experimental::optional;
...
+    return sentry_url;
+  }
   return {};
 }
 
+float get_sentry_sample_rate(
+    const magma::mconfig::SentryConfig& sentry_config,
+    YAML::Node control_proxy_config) {
+  if (control_proxy_config[SENTRY_SAMPLE_RATE].IsDefined()) {
+    const float override = control_proxy_config[SENTRY_SAMPLE_RATE].as<float>();
```
Comment (themarwhal) [inline lte/gateway/c/session_manager/SentryWrappers.cpp]: `sample_rate_override`

**Review by themarwhal — APPROVED** [review]
the changes look good thanks!

My only concern is that the sample rate from the config is 0.5 by default at the moment, we might want that value to be consistent with the default value we use here.

**Comment by waqaraqeel** [comment]
> the changes look good thanks!
> 
> My only concern is that the sample rate from the config is 0.5 by default at the moment, we might want that value to be consistent with the default value we use here.

I have changed the default `sample_rate` in `control_proxy.yml` to 1.0. Also changed the local variable names to reflect your suggestions. Thanks!

**Pull request merged**